src/cc/lib/graph/partition.cc

```diff
@@ -441,9 +441,9 @@ void Partition::GetNodeStringFeature(uint64_t internal_node_id, std::span<const
     }
 }
 
-size_t Partition::FullNeighbor(uint64_t internal_id, std::span<const Type> edge_types,
-                               std::vector<NodeId> &out_neighbors_ids, std::vector<Type> &out_edge_types,
-                               std::vector<float> &out_edge_weights) const
+template <class F>
+size_t Partition::FetchNeighborInfo(uint64_t internal_id, std::span<const Type> edge_types, F func) const
```

Comment:
Could we rename this to "NeighborIndexIterator" or something more apt, it seems this is being used more than just to get info.